### PR TITLE
chore: remove eol nodejs in tests

### PR DESCRIPTION
**Pull request by Mikescops**
As per #1886, here is a PR to drop support for 12 which is EOL and 14 which is EOL in a month.

As some users may rely on those versions, this should be included in a major release.

### CODE REVIEW

**Comment by Mikescops** [comment]
The repo owner should remove the v14 requirement for the CI so this PR can be accepted.

**Review by robertsLando — CHANGES_REQUESTED** [review]
AFAIK we planned to also keep last LTS and one EOL so I would keep the nodejs 14 for now

**Comment by baparham** [comment]
I agree with Daniel, at least for the next month, we do need to keep node 14 around until it's officially EOL. What we do after that is still up for debate I suppose, but we should be able to land this and at least drop support for 12 with a major version bump.

File: package.json [hunk 1]
```diff
@@ -78,10 +78,7 @@
     "fix": "npm run lint:style -- -w && npm run lint:code -- --fix",
     "prepare": "npm run build",
     "prepublishOnly": "npm run lint",
-    "test": "npm run build && npm run test:14 && npm run test:12 && npm run test:10 && npm run test:host",
-    "test:10": "node test/test.js node10 no-npm",
-    "test:12": "node test/test.js node12 no-npm",
-    "test:14": "node test/test.js node14 no-npm",
+    "test": "npm run build && npm run test:host",
```
Comment (baparham) [inline package.json:81]: should this be updated instead to run tests against 16, 18 and 19? I'm not exactly sure what is the goal of this existing infrastructure to test on a matrix of node versions in the runner, and run test scripts targeting specific node versions...cross-target compatibility maybe?

Comment (Mikescops) [reply package.json:81]: ah probably, i will limit it to the latest nodejs to test older versions
it will speed up things i guess

**Comment by baparham** [comment]
@robertsLando it looks like @Mikescops fixed the requested change and put node 14 back in, but this still has your requested changes flag on it

**Review by robertsLando — APPROVED** [review]
Because I didn't receive any request for new review :) BTW LGTM

**Pull request merged**